akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala

```diff
@@ -1,867 +1,603 @@
     }
 
     "set small receive timeout" in {
-      sync(setup("ctx30") { (ctx, startWith) ⇒
-        val self = ctx.self
-        startWith
-          .stimulate(_ ! SetTimeout(1.nano, self), _ ⇒ TimeoutSet)
-          .expectMessage(expectTimeout) { (msg, _) ⇒
-            msg should ===(GotReceiveTimeout)
-          }
-      })
+      val probe = TestProbe[Event]()
+      val actor = spawn(Behaviors.immutablePartial[Command] {
+        case (_, ReceiveTimeout) ⇒
+          probe.ref ! GotReceiveTimeout
+          Behaviors.same
+        case (ctx, SetTimeout(duration)) ⇒
+          ctx.setReceiveTimeout(duration, ReceiveTimeout)
+          probe.ref ! TimeoutSet
+          Behaviors.same
+      }.decorate)
+      actor ! SetTimeout(1.nano)
+      probe.expectMessage(TimeoutSet)
+      probe.expectMessage(GotReceiveTimeout)
     }
 
     "set large receive timeout" in {
-      sync(setup("ctx31") { (ctx, startWith) ⇒
-        val self = ctx.self
-        startWith
-          .stimulate(_ ! SetTimeout(1.minute, self), _ ⇒ TimeoutSet)
-          .stimulate(_ ⇒ ctx.schedule(1.second, self, Pong2), _ ⇒ Pong2)
-          .stimulate(_ ! Ping(self), _ ⇒ Pong1)
-
-      })
+      val probe = TestProbe[Event]()
+      val actor = spawn(Behaviors.immutablePartial[Command] {
+        case (ctx, Inert) ⇒
+          ctx.schedule(1.second, probe.ref, InertEvent)
+          Behaviors.same
+        case (_, Ping) ⇒
+          probe.ref ! Pong
+          Behaviors.same
+        case (_, ReceiveTimeout) ⇒
+          probe.ref ! GotReceiveTimeout
+          Behaviors.same
+        case (ctx, SetTimeout(duration)) ⇒
+          ctx.setReceiveTimeout(duration, ReceiveTimeout)
+          probe.ref ! TimeoutSet
+          Behaviors.same
+      }.decorate)
+      actor ! SetTimeout(1.minute)
+      probe.expectMessage(TimeoutSet)
+      actor ! Inert
+      probe.expectMessage(InertEvent)
+      actor ! Ping
+      probe.expectMessage(Pong)
     }
 
     "schedule a message" in {
-      sync(setup("ctx32") { (ctx, startWith) ⇒
-        startWith(_ ! Schedule(1.nano, ctx.self, Pong2, ctx.self))
-          .expectMultipleMessages(expectTimeout, 2) { (msgs, _) ⇒
-            msgs should ===(Scheduled :: Pong2 :: Nil)
-          }
-      })
-    }
-
-    "create a working adapter" in {
-      sync(setup("ctx40", ignorePostStop = false) { (ctx, startWith) ⇒
-        startWith.keep { subj ⇒
-          subj ! GetAdapter(ctx.self)
-        }.expectMessage(expectTimeout) { (msg, subj) ⇒
-          val Adapter(adapter) = msg
-          ctx.watch(adapter)
-          adapter ! Ping(ctx.self)
-          (subj, adapter)
-        }.expectMessage(expectTimeout) {
-          case (msg, (subj, adapter)) ⇒
-            msg should ===(Pong1)
-            ctx.stop(subj)
-            adapter
-        }.expectMulti(expectTimeout, 2) { (msgs, adapter) ⇒
-          msgs.toSet should ===(Set(Left(Terminated(adapter)(null)), Right(GotSignal(PostStop))))
-        }
-      })
```

Comment:
This test case wasn't ported, how come?

Reply:
Seems still the case that this wasn't ported?

Reply:
Method is still available, so I think this should be about the anonymout one, other than the one below which used a named one

```
  /**
   * Create an anonymous child actor that will wrap messages such that other Actor’s
   * protocols can be ingested by this Actor. You are strongly advised to cache
   * these ActorRefs or to stop them when no longer needed.
   */
  def spawnAdapter[U](f: U ⇒ T): ActorRef[U]

```

?

Reply:
Yep, included it as part of `"create a named adapter"` case as was proposed in first round of review below: used adapter to see that message is passed to the actor.